torch/testing/_internal/common_utils.py

```diff
@@ -682,14 +694,36 @@ def addSkip(self, test, reason):
         test_filename = sanitize_test_filename(inspect.getfile(sys._getframe(1)))
         test_report_path = TEST_SAVE_XML + LOG_SUFFIX
         test_report_path = os.path.join(test_report_path, test_filename)
-        os.makedirs(test_report_path, exist_ok=True)
-        verbose = '--verbose' in argv or '-v' in argv
-        if verbose:
-            print('Test results will be stored in {}'.format(test_report_path))
-        unittest.main(argv=argv, testRunner=xmlrunner.XMLTestRunner(
-            output=test_report_path,
-            verbosity=2 if verbose else 1,
-            resultclass=XMLTestResultVerbose))
+        if test_filename in PYTEST_FILES and not IS_SANDCASTLE and not (
+            "cuda" in os.environ["BUILD_ENVIRONMENT"] and "linux" in os.environ["BUILD_ENVIRONMENT"]
```

Comment:
Hi @malfet @janeyx99 , it seems like our `test_ops` tests were completely skipped after this PR because we don't have `BUILD_ENVIRONMENT` in our environment. Is this check really necessary for non-github-CI builds? Can you provide a fix to re-enable `test_ops` and other `PYTEST_FILES` tests?

cc @ptrblck 


```python
Running test_ops ... [2022-07-28 03:25:21.888236]
Executing ['/opt/conda/bin/python', '-bb', 'test_ops.py', '-v', '--save-xml', '--import-slow-tests', '--import-disabled-tests'] ... [2022-07-28 03:25:21.888297]
Traceback (most recent call last):
  File "test_ops.py", line 1736, in <module>
    run_tests()
  File "/opt/pytorch/pytorch/torch/testing/_internal/common_utils.py", line 716, in run_tests
    "cuda" in os.environ["BUILD_ENVIRONMENT"] and "linux" in os.environ["BUILD_ENVIRONMENT"]
  File "/opt/conda/lib/python3.8/os.py", line 675, in __getitem__
    raise KeyError(key) from None
KeyError: 'BUILD_ENVIRONMENT'
test_ops failed!
```